### PR TITLE
SI-7403 Stream extends Serializable

### PR DESCRIPTION
**Pull request by soc**
Additionally, add @deprecatedInheritance to warn creators of
custom subclasses that the class will be sealed in the future.

### CODE REVIEW

**Comment by soc** [comment]
Review by @paulp, @retronym please.

File: src/library/scala/collection/immutable/Stream.scala [hunk 1]
```diff
@@ -183,10 +183,12 @@ import scala.language.implicitConversions
  *  @define orderDependentFold
  *  @define willTerminateInf Note: lazily evaluated; will terminate for infinite-sized collections.
  */
+@deprecatedInheritance("This class will me made final.", "2.11.0")
```
Comment (paulp) [inline src/library/scala/collection/immutable/Stream.scala]: "will be"

Comment (soc) [inline src/library/scala/collection/immutable/Stream.scala]: Argh ... copy-paste error from https://github.com/scala/scala/blob/master/src/library/scala/math/BigInt.scala#L111. I'll fix this.

**Comment by paulp** [comment]
LGTM

**Comment by soc** [comment]
```
BUILD SUCCESSFUL
Total time: 20 minutes 1 second
Notifying upstream projects of job completion
Finished: SUCCESS
```

Not sure why the status doesn't update here...

**Pull request merged**